File: 프로그래머스/Level2/가장 큰 정사각형 찾기.py
# https://programmers.co.kr/learn/courses/30/lessons/12905?language=python3

def solution(board):
    row = len(board)
    col = len(board[0])
    for i in range(1, row):
        for j in range(1, col):
            if(board[i][j] != 0):
                board[i][j] = min(board[i-1][j-1], board[i][j-1], board[i-1][j]) + 1
    size = 0
    for arr in board:
        size = max(size, max(arr))
    return size ** 2

File: 프로그래머스/Level2/가장 큰 정사각형 찾기.js
// https://programmers.co.kr/learn/courses/30/lessons/12905

const solution = (board) => {
	const row = board.length;
	const col = board[0].length;
	for (let i = 1; i < row; i++) {
		for (let j = 1; j < col; j++) {
			if (board[i][j] !== 0)
				board[i][j] = Math.min(board[i - 1][j - 1], board[i][j - 1], board[i - 1][j]) + 1;
		}
	}
	let size = 0;
	board.forEach((arr) => {
		size = Math.max(...arr, size);
	});
	return size ** 2;
};
